src/rules/extensions.js

```diff
@@ -37,15 +39,15 @@ module.exports = function (context) {
     const extension = path.extname(resolvedPath || importPath).substring(1)
 
     if (!extension || !endsWith(importPath, extension)) {
-      if (isUseOfExtensionEnforced(extension)) {
+      if (isUseOfExtensionRequired(extension) && !isUseOfExtensionForbidden(extension)) {
```

Comment:
So, I'm not anti-dependencies in general, but I don't understand why 

- someone would have file extensions of `toString` or `hasOwnProperty`
- someone would be surprised if things are broken when they monkeypatch JS core functions

Am I correct that these are the cases where `has` matters over just a plain ol' `hasOwnProperty` call or `in` operator? Because if that's true, I'd still like to revert to `hasOwnProperty` or really, just `extension in modifiers` because these are well-understood JS core idioms, and in this specific case, not practically distinguishable.